wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/environment/server/compose/rule/CommandDeserializer.java

```diff
@@ -0,0 +1,110 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.api.environment.server.compose.rule;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.TreeNode;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+
+import org.eclipse.che.api.environment.server.compose.ComposeServiceImpl;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.lang.String.format;
+import static java.util.Arrays.asList;
+
+/**
+ * Custom deserializer for parsing {@link ComposeServiceImpl#command}
+ * to list command words in the yaml compose file.
+ * <p>Command can be:</p>
+ * <ul>
+ * <li>string array:</li>
+ *
+ * <pre>
+ *  image: codenvy/mysql
+ *  environment:
+ *   MYSQL_USER: petclinic
+ *   MYSQL_PASSWORD: password
+ *  mem_limit: 2147483648
+ *  command: [service, mysql, start]
+ * </pre>
+ *
+ * <li>simple string line:</li>
+ *
+ * <pre>
+ *  image: codenvy/mysql
+ *  environment:
+ *   MYSQL_USER: petclinic
+ *   MYSQL_PASSWORD: password
+ *  mem_limit: 2147483648
+ *  command: service mysql start
+ * </pre>
+ * </ul>
+ *
+ *
+ * See more for parsing compose file
+ * {@link org.eclipse.che.api.environment.server.compose.ComposeFileParser}.
+ * <p>Note: this deserializer works for json too.</p>
+ *
+ * @author Alexander Andrienko
+ */
+public class CommandDeserializer extends JsonDeserializer<List<String>> {
+
+    private static final String SPLIT_COMMAND_REGEX = "[ ]+";
+
+    /**
+     * Parse command field from the compose yaml file to list command words.
+     *
+     * @param jsonParser
+     *         json parser
+     * @param ctxt
+     *         deserialization context
+     * @throws IOException
+     *         in case I/O error. For example element to parsing contains
+     *         invalid yaml field type defined for this field by yaml document model.
+     * @throws JsonProcessingException
+     */
+    @Override
+    public List<String> deserialize(JsonParser jsonParser, DeserializationContext ctxt) throws IOException, JsonProcessingException {
```

Comment:
Provided by the documentation example:

```yaml
  image: codenvy/mysql
  environment:
    MYSQL_USER: petclinic
    MYSQL_PASSWORD: password
  mem_limit: 2147483648
  command: service mysql start
```
it is clear yaml format, why do you use _JSON_ parser for parsing command line from yaml format?

As far as i understand you should add support for `command: service mysql start` when this command is parsed from yaml to object by `ComposeFileParser`, do i miss something?